source/worksheet/worksheet.cpp

```diff
@@ -736,6 +740,128 @@ void worksheet::clear_row(row_t row)
     // TODO: garbage collect newly unreferenced resources such as styles?
 }
 
+void worksheet::insert_rows(row_t row, std::uint32_t amount)
+{
+    move_cells(row, amount, row_or_col_t::row);
+}
+
+void worksheet::insert_columns(column_t column, std::uint32_t amount)
+{
+    move_cells(column.index, amount, row_or_col_t::column);
+}
+
+void worksheet::delete_rows(row_t row, std::uint32_t amount)
+{
+    move_cells(row + amount, amount, row_or_col_t::row, true);
+}
+
+void worksheet::delete_columns(column_t column, std::uint32_t amount)
+{
+    move_cells(column.index + amount, amount, row_or_col_t::column, true);
+}
+
+void worksheet::move_cells(std::uint32_t min_index, std::uint32_t amount, row_or_col_t row_or_col, bool reverse)
```

Comment:
I'm adding a check, but given the line above it it is not possible to have min_index < amount as min_index is the column/row index + amount. This should only become relevant if the `move_cells()` method is used elsewhere.

Reply:
On the other hand, it might be worth adding max boundary checks to make sure you don't insert past `max_row` and `max_column`. Is that something generally handled?

Reply:
Not sure if it would be worth it. The min/max column/row functions are currently linear searches (should really be cached) :/